course.go

```diff
@@ -32,6 +32,18 @@ type CourseUpdater interface {
 	Update(id string, update bson.M) (error, Course)
 }
 
+type CoursesByMemberFinder interface {
```

Comment:
The downside with update is, that we need to use the mgo package inside of the service layer.
Is there a way to make the interface smaller without using bson.M in the service layer?

Reply:
I was thinking about that. Conclusion was that Go actually has no package hierarchy. So `mgo` is not really the same as `bson` (eventhough they are shipped together). I had the same conflict of mind when adding `bson.ObjectID`s to the domain models. In the end I think its not so bad, that services are defining queries and repositories making sure they get executed correctly. Another solution to this would be to accept `map[string]interface{}` which is type aliased by `bson.M` for convenience. The repository could cast it to the bson type then.